Make sys_poll() wait at least timeout ms
From: Karsten Wiese <fzu@wemgehoertderstaat.de>

schedule_timeout(jiffies) waits for at least jiffies - 1.
Add 1 jiffie to the timeout_jiffies calculated in sys_poll() to wait at least
timeout_msecs, like poll() manpage says.

Signed-off-by: Karsten Wiese <fzu@wemgehoertderstaat.de>
---

 fs/select.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.24.7-rt27/fs/select.c
===================================================================
--- linux-2.6.24.7-rt27.orig/fs/select.c	2009-02-08 00:00:39.000000000 -0500
+++ linux-2.6.24.7-rt27/fs/select.c	2009-02-08 00:00:42.000000000 -0500
@@ -739,7 +739,7 @@ asmlinkage long sys_poll(struct pollfd _
 			timeout_jiffies = -1;
 		else
 #endif
-			timeout_jiffies = msecs_to_jiffies(timeout_msecs);
+			timeout_jiffies = msecs_to_jiffies(timeout_msecs) + 1;
 	} else {
 		/* Infinite (< 0) or no (0) timeout */
 		timeout_jiffies = timeout_msecs;
